src/LibraryManager.Vsix/Shared/Telemetry.cs

```diff
@@ -46,5 +48,96 @@ public static void TrackException(string name, Exception exception)
             string actualName = name.Replace(" ", "_");
             TelemetryService.DefaultSession.PostFault(_namespace + actualName, exception.Message, exception);
         }
+
+        internal static void LogEventsSummary(IEnumerable<ILibraryOperationResult> results, OperationType operation, TimeSpan elapsedTime)
+        {
+            double elapsedTimeRounded = Math.Round(elapsedTime.TotalSeconds, 2);
+            string elapsedTimeStr = elapsedTimeRounded.ToString(System.Globalization.CultureInfo.InvariantCulture);
+            List<string> generalErrorCodes = GetErrorCodes(results.Where(r => r.InstallationState == null && r.Errors.Any()));
+
+            Dictionary<string, object> telResult = new Dictionary<string, object>();
+            telResult.Add("LibrariesCount", results.Count());
+            telResult.Add($"{operation}_time", elapsedTimeStr);
+            telResult.Add("ErrorCodes", string.Join(" :", generalErrorCodes));
+
+            IEnumerable<string> providers = results.Select(r => r.InstallationState?.ProviderId).Distinct();
+
+            foreach (string provider in providers)
+            {
+                IEnumerable<ILibraryOperationResult> providerResults = results.Where(r => r.InstallationState?.ProviderId == provider);
+                List<string> providerErrorCodes = GetErrorCodes(providerResults);
+
+                List<string> librariesNames_Success = GetLibrariesNames(providerResults.Where(r => r.Success && !r.UpToDate));
+                List<string> librariesNames_Failure = GetLibrariesNames(providerResults.Where(r => r.Errors.Any()));
+                List<string> librariesNames_Cancelled = GetLibrariesNames(providerResults.Where(r => r.Cancelled));
+                List<string> librariesNames_Uptodate = GetLibrariesNames(providerResults.Where(r => r.UpToDate));
+
+                telResult.Add($"ErrorCodes_{provider}", string.Join(" :", providerErrorCodes));
+
+                if (librariesNames_Success.Count > 0)
+                {
+                    telResult[$"LibrariesCount_{provider}_Success"] = librariesNames_Success.Count;
+                    telResult[$"LibrariesIDs_{provider}_Success"] = new TelemetryPiiProperty(string.Join(" :", librariesNames_Success));
```

Comment:
>new TelemetryPiiProperty [](start = 68, length = 24)

We need to decide if this hashed data will be useful. Might be better to log library IDs on the specific failure events, and just leave the counts here.